### PR TITLE
[Term Entry] Queue for Dart

### PR DESCRIPTION
**Pull request by KrisHaroun**
### Description

- Added the Queue term for Dart
- Closes issue #4241 

### Issue Solved

- Closes issue #4241 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/queue/queue.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Error: Method not found: 'Queue'.
+  Queue<String> example_queue = new Queue<String>();
+                                    ^^^^^
+Error: Compilation failed.
+```
+## Example
+
+In the following example, a queue is created and elements are inserted in the queue using the **.add()** method.
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:47]:
```suggestion
```dart
```

File: content/dart/concepts/queue/queue.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   
+  // Adding elements in a Queue
+  geek.add("This");
+  geek.add("Is");
+  geek.add("A");
+  geek.add("Queue");
+   
+  // Printing the new updated queue
+  print(example_queue);
+}
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:66]:
```suggestion
import 'dart:collection';
 
void main()
{
  // Creating a Queue
  Queue<String> example_queue = new Queue<String>(); 
   
  // Printing default value of queue
  print(example_queue);
   
  // Adding elements in a Queue
  example_queue .add("This");
  example_queue .add("Is");
  example_queue .add("A");
  example_queue .add("Queue");
   
  // Printing the new updated queue
  print(example_queue);
}
```
The name of the queue was incorrect.

File: content/dart/concepts/queue/queue.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+```
+Output:
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:68]:
```suggestion

The above code will result in the following output:
```

File: content/dart/concepts/queue/queue.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:70]:
```suggestion
```shell
```

File: content/dart/concepts/queue/queue.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Queue variable_name = new Queue();
+```
+You can also create a queue from an existing list:
+
+```
+var variable_name = new Queue.from(list_name);
+```
+It must be noted that to use a queue in a dart program you have to import "dart:collection" module. If you don’t do so then you will see the following error:
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:31]:
```suggestion
```pseudo
```

File: content/dart/concepts/queue/queue.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  - 'paths/computer-science'
+---
+
+A **Queue** is a collection that can be manipulated at both ends. A Queue inserts data from one end and deletes data from another end, it is considered a "first in first out" collection.
+
+## Syntax
+
+Creating a queue:
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:21]:
```suggestion
```pseudo
```
The syntax should be enclosed within the ` ```pseudo` block.

File: content/dart/concepts/queue/queue.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+Queue variable_name = new Queue();
+```
+You can also create a queue from an existing list:
+
+```
+var variable_name = new Queue.from(list_name);
+```
+It must be noted that to use a queue in a dart program you have to import "dart:collection" module. If you don’t do so then you will see the following error:
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:29]: Give one line space before and after a code block. You can add this line as a note. Here's an example of a note - 
> **Note:** Note Text.

File: content/dart/concepts/queue/queue.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+A **Queue** is a collection that can be manipulated at both ends. A Queue inserts data from one end and deletes data from another end, it is considered a "first in first out" collection.
+
+## Syntax
+
+Creating a queue:
+
+```
+Queue variable_name = new Queue();
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:22]:
```suggestion
Queue queue_name = new Queue();
```

File: content/dart/concepts/queue/queue.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+```
+You can also create a queue from an existing list:
+
+```
+var variable_name = new Queue.from(list_name);
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:27]:
```suggestion
var queue_name = new Queue.from(list_name);
```

File: content/dart/concepts/queue/queue.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+A **Queue** is a collection that can be manipulated at both ends. A Queue inserts data from one end and deletes data from another end, it is considered a "first in first out" collection.
+
+## Syntax
+
+Creating a queue:
+
+```
+Queue variable_name = new Queue();
+```
+You can also create a queue from an existing list:
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:24]:
```suggestion

You can also create a queue from an existing list:
```

File: content/dart/concepts/queue/queue.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  - 'Computer Science'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Classes'
+  - 'Queues'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+A **Queue** is a collection that can be manipulated at both ends. A Queue inserts data from one end and deletes data from another end, it is considered a "first in first out" collection.
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:15]: This line contains plagiarism, please change it.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@KrisHaroun Hey, please make these changes.

File: content/dart/concepts/queue/queue.md (hunk 7 above)
Comment (KrisHaroun) [reply content/dart/concepts/queue/queue.md:30]: Hello, I commited all the other changes except this one. I'm not really sure what this means, do you want me to add an empty note line before and after each code block?

File: content/dart/concepts/queue/queue.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+Error: 'Queue; isn't a type
+  Queue<String> example_queue = new Queue<String>();
+  ^^^^^
+main.dart:6:28:
+Error: Method not found: 'Queue'.
+  Queue<String> example_queue = new Queue<String>();
+                                    ^^^^^
+Error: Compilation failed.
+```
+## Example
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:44]:
```suggestion

## Example
```

File: content/dart/concepts/queue/queue.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+}
+```
+
+The above code will result in the following output:
+
+```shell
+{}
+{This, Is, A, Queue}
+```
+
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:76]:
```suggestion
```

File: content/dart/concepts/queue/queue.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
+---
+Title: 'Queue'
+Description: 'A collection that can be manipulated at both ends.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:7]: Remove the commented line.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Please resolve all of the suggested changes.

File: content/dart/concepts/queue/queue.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+Creating a queue:
+
+```pseudo
+Queue queue_name = new Queue();
+```
+
+You can also create a queue from an existing list:
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:27]:
```suggestion
```pseudo
```

File: content/dart/concepts/queue/queue.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+Queue queue_name = new Queue();
+```
+
+You can also create a queue from an existing list:
+
+```pseudo
+var queue_name = new Queue.from(list_name);
+```
+
+**Note:** To use a queue in a dart program you have to import "dart:collection" module. If you don’t do so then you will see the following error:
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:31]:
```suggestion
> **Note:** To use a queue in a dart program you have to import `dart:collection` module. If you don’t do so then you will see the following error:
```

File: content/dart/concepts/queue/queue.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+main.dart:6:28:
+Error: Method not found: 'Queue'.
+  Queue<String> example_queue = new Queue<String>();
+                                    ^^^^^
+Error: Compilation failed.
+```
+
+## Example
+
+In the following example, a queue is created and elements are inserted in the queue using the **.add()** method.
```
Comment (avdhoottt) [inline content/dart/concepts/queue/queue.md:48]:
```suggestion
In the following example, a queue is created and elements are inserted in the queue using the `.add()` method.
```

**Review by avdhoottt — APPROVED** [review]
Looks good for the second review.

**Review by Sriparno08 — APPROVED** [review]
Looks good 👍

**Pull request merged**